ray-operator/controllers/raycluster/common/pod.go

```diff
@@ -45,16 +47,8 @@ func DefaultHeadPodTemplate(instance rayiov1alpha1.RayCluster, headSpec rayiov1a
 		// set custom service account with proper roles bound.
 		podTemplate.Spec.ServiceAccountName = utils.GetHeadGroupServiceAccountName(&instance)
 
-		// Note: Starting with the upcoming Ray 1.11.0, Ray will by default no longer use Redis
```

Comment:
instead of removing it, can we check ray.spec.version make it compatible with earlier versions?

Reply:
Ray autoscaler support in KubeRay is actually not at all compatible with Ray versions < 1.11.0 (the Ray versions that use Redis).

And it looks like we were only using the Redis password to format the autoscaler container...so seems we don't need it.

I guess we should handle older Ray versions in some way -- if ray.spec.version < 1.11.0, log a warning and don't add the container? What do you think?

Reply:
> Ray autoscaler support in KubeRay is actually not at all compatible with Ray versions < 1.11.0 (the Ray versions that use Redis).

This is true.  The problem is we are still using older version without autoscaling support in downstream. If we cherry-pick this change, I think it will break. Can we add the if else logic and document the context?

Reply:
Hmm, trying to understand --

Currently we have the line
`container := BuildAutoscalerContainer(redisPasswd)`
the redisPasswd argument is ultimately ignored because the autoscaler does not connect to a redis server.

Is the issue with removing the following lines?
```
		// should be possible to drop some of the logic around Redis passwords at that point.
		// TODO(jiaxin.shan): Add version compatibility for 1.11.0 later.
		redisPasswd := instance.Spec.HeadGroupSpec.RayStartParams["redis-password"]
		if len(redisPasswd) == 0 {
			redisPasswd = DefaultRedisPassword
		}
```

Is `redisPasswd` used in the downstream in a way that it isn't used here, such that removing those lines of code would break downstream after a cherry-pick?

Is that possible to fix in the downstream?

Reply:
Chatted offline to clear this up --
The logic being removed here is only used in formatting the head pod's autoscaler. Worker pod connection logic using redis password is left untouched. So we should be good here!